src/core/render/compiler.js

```diff
@@ -235,7 +236,12 @@ export class Compiler {
       const slug = slugify(config.id || str);
       const url = router.toURL(router.getCurrentPath(), { id: slug });
       nextToc.slug = url;
-      _self.toc.push(nextToc);
+
+      // FIXME: Maybe we can do the de-duplication directly from _self.toc
+      if (_self.uniqueTOC.indexOf(url) === -1) {
+        _self.uniqueTOC.push(url);
+        _self.toc.push(nextToc);
+      }
```

Comment:
It is not clear what this changes or how the original problem happens. Can you explain?

Reply:
you can see https://discord.com/channels/713647066802421792/713647159882547261/813714253743128606

Reply:
You only pointed to some code. Can you explain?

media files should not have anything to do with the sidebar, so it **_seems_** this is not the correct fix, and that we should fix a problem with embedding media (or other stuff too maybe).

Reply:
What I mean is, that fact that this code seems entirely unrelated to media files, yet fixes the problem, seems more like a workaround than an actual fix.

firstly, why does adding media files cause duplication?